benchmarks/benchmarks/stats.py

```diff
@@ -150,3 +150,19 @@ def time_ortho_group(self, dim):
 
     def time_special_ortho_group(self, dim):
         stats.special_ortho_group.rvs(dim)
+
+
+class BinnedStatistic(Benchmark):
+
+    def setup(self):
+        np.random.seed(12345678)
+        self.inp = np.random.rand(9999).reshape(3, 3333) * 200
+        self.subbin_x_edges = np.arange(0, 200, dtype=np.float32)
+        self.subbin_y_edges = np.arange(0, 200, dtype=np.float64)
+
+    def time_binned_statistic_2d_std(self):
+        stat = 'std'
```

Comment:
This is a useful change to track, thanks for adding the benchmark. 
The `binned_statistic_dd` method supports both a scenario where you pass in data and the bin edges are calculated inside the method as well as the scenario where you've already computed the binedges and would like to re-use them. It would be nice to track both input cases for this method. Tracking both situations will give an indication of the speed up due to changes in std calculation alone versus other changes inside the routine.

Reply:
OK. I will try it.

Reply:
I tried use `binned_statistic_result` for reused benchmark. But, an error happened and I don't know how to fix it... Could you please help me?

This is my code.
```
import numpy as np
from scipy import stats

np.random.seed(12345678)
inp = np.random.rand(9999).reshape(3, 3333) * 200
subbin_x_edges = np.arange(0, 200, dtype=np.float32)
subbin_y_edges = np.arange(0, 200, dtype=np.float64)
stat = 'std'

ret = stats.binned_statistic_dd(
    [inp[0], inp[1]], inp[2], statistic=stat,
    bins=[subbin_x_edges, subbin_y_edges],
    expand_binnumbers=True)
print(ret)

ret = stats.binned_statistic_dd(
    [inp[0], inp[1]], -inp[2], statistic=stat,
    binned_statistic_result=ret)
print(ret)
```

But, this error happened in the second `binned_statistic_dd` call.

>  File "/home/scipy/build/testenv/lib/python3.6/site-packages/scipy/stats/_binned_statistic.py", line 584, in binned_statistic_dd
>                    flatcount = np.bincount(binnumbers, None)
>                  File "<__array_function__ internals>", line 6, in bincount
>                ValueError: object too deep for desired array

Reply:
>It would be nice to track both input cases for this method.

I added re-use bin benchmark too.